docs/pnpm.md

```diff
@@ -0,0 +1,282 @@
+# pnpm and rules_js
+
+rules_js models dependency handling on [pnpm](https://pnpm.io/). Our design goal is to closely mimic pnpm's behavior.
+
+To start, some non-Bazel-specific tool (typically pnpm) which performs dependency resolutions
+and solves version constraints. It also determines how the `node_modules` tree will be structured for runtime.
+This information is encoded into a lockfile which is checked into the source repository.
+
+The pnpm lockfile format includes all the information needed to define [`npm_import`](/docs/npm_import.md#npm_import) rules,
+allowing Bazel's downloader to do the fetches. This info includes the integrity hash, as calculated by the package manager,
+so that Bazel can guarantee supply-chain security.
+
+For more details, see <https://github.com/pnpm/pnpm/blob/main/packages/lockfile-types/src/index.ts>.
```

Comment:
https://github.com/pnpm/pnpm/blob/main/packages/lockfile-types/src/index.ts gives me a 404

Reply:
I wonder where we even got that line from, it's been in the docs a long time

Reply:
just removing the link altogether